Hazel/src/Hazel/Core/EntryPoint.h

```diff
@@ -1,8 +1,9 @@
 #pragma once
+#include "Hazel/Core/Core.h"
 
 #ifdef HZ_PLATFORM_WINDOWS
 
-extern Hazel::Application* Hazel::CreateApplication();
+extern Hazel::Scope<Hazel::Application> Hazel::CreateApplication();
```

Comment:
I remember that Cherno wanted full control over when application gets closed.

Reply:
Ok, we can revert this one back to a raw pointer.

Reply:
Resolved in https://github.com/TheCherno/Hazel/pull/165/commits/47e7ebf8f1fe6d56aaa93456fc099b014e94f821

Reply:
I think making it a smart pointer is actually fine. If we look at how `EntryPoint` works we see that the entire lifetime of the application is spent in the `Application::Run` method which means we can never call `delete app` in the application because we are freeing memory that is still being used, which would then cause a crash.

The mechanism for closing the application is to set `Application::m_Running` to false since that ends the while loop in the run method and then the next line of code that gets executed in `EntryPoint` is to delete the Application. This is also what we currently do in `Application::OnWindowClose`. The change to a smart pointer still allows this method of closing the application and simply removes the need for that `delete` call in `EntryPoint`.

Closing the application through this mechanism also ensures that we don't close mid-render or mid-some-other-process and allow all processes to finish normally before exiting.

Also no part of the application has access to the pointer returned by this starting method anyway, since it only exists in `EntryPoint`. You might also note that we do keep `Application::s_Instance` a raw pointer, but this is because `Application` cannot have ownership of itself, which is what a `Scope` would imply.